engine/src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -56,67 +55,80 @@ class ValidateActor()
+
+      case scalaz.Failure(failure) => failure.failure
+    }
+
+    result match {
+      case scalaz.Success(success) => success
+      case scalaz.Failure(failure) =>
+        ValidationFailure(new ValidationException(s"Workflow input processing failed.", failure))
     }
   }
 
-  // TODO: With PBE, this should be defined in the backend.
-  private def validateRuntimeAttributes(namespaceWithWorkflow: NamespaceWithWorkflow): Seq[Set[String]] = {
-    TryUtil.sequence(namespaceWithWorkflow.workflow.calls map {
-      call => CromwellRuntimeAttributes.validateKeys(call.task.runtimeAttributes.attrs.keySet, CromwellBackend.backend().backendType)
-    }) match {
-      case Success(validatedRuntimeAttrs) => validatedRuntimeAttrs
-      case Failure(reason) => throw new IllegalArgumentException("Failed to validate runtime attributes.", reason)
+  private def validateNamespace(source: WdlSource): ErrorOr[NamespaceWithWorkflow] = {
+    try {
+      NamespaceWithWorkflow.load(source).successNel
+    } catch {
+      case e: Exception => s"Unable to load namespace from workflow: ${e.getMessage}".failureNel
     }
   }
 
-  private def validateInputs(workflowInputs: Option[WdlJson], namespaceWithWorkflow: NamespaceWithWorkflow): Option[WorkflowCoercedInputs] = {
-    def rawInputs(json: WdlJson): Map[String, JsValue] = {
+  private def validateInputs(namespaceWithWorkflow: NamespaceWithWorkflow, workflowInputs: Option[WdlJson]): ErrorOr[WorkflowCoercedInputs] = {
+    def validateRawInputs(json: WdlJson): ErrorOr[Map[String, JsValue]] = {
       Try(json.parseJson) match {
-        case Success(JsObject(inputs)) => inputs
-        case Failure(reason: Throwable) => throw new Throwable(reason.getMessage)
-        case _ => throw new IllegalArgumentException(s"Failed to parse the workflow inputs JSON: $json")
+        case Success(JsObject(inputs)) => inputs.successNel
+        case Failure(reason: Throwable) => s"Workflow contains bad inputs JSON: ${reason.getMessage}".failureNel
+        case _ => s"Workflow contains bad inputs JSON: $json".failureNel
       }
     }
 
-    workflowInputs match {
-      case Some(json) =>
-        namespaceWithWorkflow.coerceRawInputs(rawInputs(json)) match {
-          case Success(coercedInputs) => Option(coercedInputs)
-          case Failure(reason: ThrowableWithErrors) => throw new ValidationException(reason.message, reason.errors)
-          case Failure(reason) => throw new IllegalArgumentException("Failed to coerce inputs", reason)
+    validateRawInputs(workflowInputs.getOrElse(EmptyJson)) match {
+      case scalaz.Success(inputs) =>
+        namespaceWithWorkflow.coerceRawInputs(inputs) match {
+          case Success(r) => r.successNel
+          case Failure(e: ThrowableWithErrors) => scalaz.Failure(e.errors)
+          case Failure(e) => e.getMessage.failureNel
         }
-      case _ => None
+      case scalaz.Failure(f) => f.failure
```

Comment:
The goal of this PR is just to integrate Scalaz lib into ValidateActor. @gauravs90 will add missing things in the next PR based on what we discussed today.